modules/base/dashboard/dashboarditemtimevaryingtext.cpp

```diff
@@ -0,0 +1,161 @@
+/*****************************************************************************************
+ *                                                                                       *
+ * OpenSpace                                                                             *
+ *                                                                                       *
+ * Copyright (c) 2014-2025                                                               *
+ *                                                                                       *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this  *
+ * software and associated documentation files (the "Software"), to deal in the Software *
+ * without restriction, including without limitation the rights to use, copy, modify,    *
+ * merge, publish, distribute, sublicense, and/or sell copies of the Software, and to    *
+ * permit persons to whom the Software is furnished to do so, subject to the following   *
+ * conditions:                                                                           *
+ *                                                                                       *
+ * The above copyright notice and this permission notice shall be included in all copies *
+ * or substantial portions of the Software.                                              *
+ *                                                                                       *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#include <modules/base/dashboard/dashboarditemtimevaryingtext.h>
+
+#include <fstream>
+#include <openspace/documentation/documentation.h>
+#include <openspace/documentation/verifier.h>
+#include <openspace/engine/globals.h>
+#include <openspace/util/timemanager.h>
+#include <openspace/json.h>
+#include <ghoul/filesystem/filesystem.h>
+#include <ghoul/logging/logmanager.h>
+
+
+
+namespace {
+    constexpr std::string_view _loggerCat = "DashboardItemTimeVaryingText";
+
+    constexpr openspace::properties::Property::PropertyInfo TextBeforeInfo = {
+        "TextBefore",
+        "Text Before",
+        "Optional text before the time varying text to be displayed.",
+        openspace::properties::Property::Visibility::User
+    };
```

Comment:
To be more consistent with other Dashboard items (DashboardItemDate, DashboardItemDistance, etc) it would be better to have this as a `FormatString` instead.  So in the example above it would be `"Observed KP index: {}"` instead. That would allow a user to also add a suffix through `"{} is the observed KP index"` for example.

You can lift the details straight from the DashboardItemDate.
Property: https://github.com/OpenSpace/OpenSpace/blob/master/modules/base/dashboard/dashboarditemdate.cpp#L39-L46
Parameters: https://github.com/OpenSpace/OpenSpace/blob/master/modules/base/dashboard/dashboarditemdate.cpp#L63-L64
Resolving the format string: https://github.com/OpenSpace/OpenSpace/blob/master/modules/base/dashboard/dashboarditemdate.cpp#L103-L109